From 9d1358ab007ef1e9a36caf952e66102cf49d7b25 Mon Sep 17 00:00:00 2001
From: Nicolas Le Bayon <nicolas.le.bayon@st.com>
Date: Tue, 19 Dec 2023 22:10:20 +0100
Subject: [PATCH 521/954] feat(stm32mp21): no GPU BW limiter disable on
 STM32MP21

GPU is not present on STM32MP21, so don't manage its BW limiter.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git af832318252b7d5ec410e12deb11461eb6394ed4 feat(stm32mp21): no GPU BW limiter disable on STM32MP21]
Signed-off-by: Nicolas Le Bayon <nicolas.le.bayon@st.com>
Change-Id: Ifdfc203df36a4b4591b9e0a61864ce9b361d5acb
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/356310
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
---
 plat/st/stm32mp2/stm32mp2_syscfg.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/plat/st/stm32mp2/stm32mp2_syscfg.c b/plat/st/stm32mp2/stm32mp2_syscfg.c
index aca04ffcbc..1a114b67f2 100644
--- a/plat/st/stm32mp2/stm32mp2_syscfg.c
+++ b/plat/st/stm32mp2/stm32mp2_syscfg.c
@@ -535,8 +535,10 @@ void stm32mp_syscfg_set_icn_qos(void)
 	/* Enable BW regulator */
 	mmio_write_32(SYSCFG_BASE + SYSCFG_ICNE2EBWRCR, U(0x00000051));
 
+#if !STM32MP21
 	/* Disable GPU BW limiter */
 	mmio_write_32(SYSCFG_BASE + SYSCFG_ICNGPUBWLCR, U(0x00004009));
+#endif /* !STM32MP21 */
 
 #if STM32MP_DDR4_TYPE || STM32MP_LPDDR4_TYPE
 	/* Change value of CPU BW limiter */
-- 
2.53.0

